dask/dataframe/io/tests/test_parquet.py

```diff
@@ -3853,3 +3853,30 @@ def _prep(x):
     df_read_dask = dd.read_parquet(pa_path, engine=engine)
     df_read_pa = pq.read_table(pa_path).to_pandas()
     assert_eq(_prep(df_read_dask), _prep(df_read_pa), check_index=False)
+
+
+@pytest.mark.parametrize("filter_value", ({1}, [1], (1,)), ids=("set", "list", "tuple"))
+def test_in_predicate_can_use_iterables(tmpdir, engine, filter_value):
+    """Fixes GH8720"""
+    path = os.path.join(str(tmpdir), "in_predicate_iterable.parquet")
+    df = pd.DataFrame(
+        {"A": [1, 2, 3, 4], "B": [1, 1, 2, 2]},
+    )
+    df.to_parquet(path, engine=engine)
+    filters = [("B", "in", filter_value)]
+    result = dd.read_parquet(path, engine=engine, filters=filters)
+    expected = pd.read_parquet(path, engine=engine, filters=filters)
+    assert_eq(result, expected)
+
+
+def test_in_predicate_requires_an_iterable(tmpdir, engine):
+    """Fixes GH8720"""
+    path = os.path.join(str(tmpdir), "gh_8720.parquet")
+    df = pd.DataFrame(
+        {"A": [1, 2, 3, 4], "B": [1, 1, 2, 2]},
+    )
+    df.to_parquet(path, engine=engine)
+    # Non-iterable filter value
+    filters = [("B", "in", 10)]
+    with pytest.raises(TypeError, match="Value of 'in' filter"):
+        dd.read_parquet(path, engine=engine, filters=filters, gather_statistics=True)
```

Comment:
Yes. For pyarrow, letting `gather_statistics` be the default value results in it being set to `False`, since statistics are not required for filtering. Therefore, it follows the shortcut path here https://github.com/dask/dask/blob/723e0d223c2921e30d4324cac60d2256ad13386b/dask/dataframe/io/parquet/arrow.py#L1180-L1188

which is immediately prior to the filtering error. `fastparquet` does not require this setting.

Reply:
Another option is to set `split_row_groups=True`. Another option is to generate a directory of parquet files _with_ a `_metadata` file instead of a single file, and set `ignore_metadata_file=False` (which I think is the default); this would set `gather_statistics=True` here https://github.com/dask/dask/blob/723e0d223c2921e30d4324cac60d2256ad13386b/dask/dataframe/io/parquet/arrow.py#L792-L850

Reply:
I guess this also means that the test above this one isn't actually testing the logic we want? At least insofar as it's not running past the `TypeError`, at least for pyarrow.

Reply:
OK, this is resolved now, I think. Sometimes the smallest questions... See the next review comment for the resolution 👇